lib/generator.dart

```diff
@@ -115,97 +115,108 @@ QueryDefinition generateQuery(
     throw FragmentIgnoreException();
   }
 
-  if (fragmentsCommon.isEmpty) {
-    fragments.addAll(documentFragments);
-  } else {
-    final fragmentsOperation =
-        _extractFragments(operation.selectionSet, fragmentsCommon);
-    document.definitions.addAll(fragmentsOperation);
-    fragments.addAll(fragmentsOperation);
-  }
+  final operations =
+      document.definitions.whereType<OperationDefinitionNode>().toList();
 
-  final basename = p.basenameWithoutExtension(path).split('.').first;
-  final operationName = operation.name?.value ?? basename;
-
-  final schemaVisitor = SchemaDefinitionVisitor();
-  final objectVisitor = ObjectTypeDefinitionVisitor();
-
-  schema.accept(schemaVisitor);
-  schema.accept(objectVisitor);
-
-  String suffix;
-  switch (operation.type) {
-    case OperationType.subscription:
-      suffix = 'Subscription';
-      break;
-    case OperationType.mutation:
-      suffix = 'Mutation';
-      break;
-    case OperationType.query:
-    default:
-      suffix = 'Query';
-      break;
-  }
+  return operations.map((operation) {
```

Comment:
Can we extract this to another function? Maybe a `generateQueries` and another `generateQuery`.